modules/javafx.base/src/main/java/com/sun/javafx/binding/ListenerManagerBase.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javafx.binding;
+
+import javafx.beans.InvalidationListener;
+import javafx.beans.value.ChangeListener;
+import javafx.beans.value.ObservableValue;
+
+/**
+ * Base class for managing a single data field of type {@link Object} to store zero,
+ * one or more {@link InvalidationListener}s and {@link ChangeListener}s. This
+ * helps to minimize the storage requirements for keeping track of these
+ * listeners.
+ *
+ * @param <T> the type of the values
+ * @param <I> the type of the instance providing listener data
+ */
+public abstract class ListenerManagerBase<T, I extends ObservableValue<? extends T>> {
```

Comment:
Is this a class and not an interface because of the need for `protected` methods?

Reply:
I think I extracted this when I discovered that I needed two implementations (caching one and one that doesn't need caching, to save more memory).

Looking at it now (over a year later) I suppose it could also be an interface.  There is no risk that any of this gets exposed as the property classes will all create a private class to implement the abstract methods.

It can be changed at any time as it is internal.  If we want to make it an interface, I'm going to need to think of a good name for it (`ListenerDataProvider`, `ListenerDataStore`, ... )

Reply:
Just making sure you didn't forget to consider this. Fine to leave as is also.

Reply:
I'll leave it if you don't mind, as said, we're not locked in for now.